### PR TITLE
Add some tests. Remove unnecessary if __name__ == __main__

### PR DESCRIPTION
**Pull request by Joao-Dionisio**

### CODE REVIEW

File: tests/test_pricer.py [hunk 1]
```diff
@@ -10,6 +10,7 @@ def pricerredcost(self):
         dualSolutions = []
         for i, c in enumerate(self.data['cons']):
             dualSolutions.append(self.model.getDualsolLinear(c))
+            #assert self.model.getDualsolLinear(c) == self.model.getDualSolVal(c)
```
Comment (mmghannam) [inline tests/test_pricer.py:13]: Why is this commented out?

File: tests/test_reader.py [hunk 2]
```diff
@@ -59,7 +59,7 @@ def createFile(filename):
 def deleteFile(filename):
     os.remove(filename)
 
-def test():
+def test_main():
```
Comment (mmghannam) [inline tests/test_reader.py:62]: Shouldn't this be something more descriptive, like `test_sudoko_reader` for example.

File: tests/test_solution.py [hunk 3]
```diff
@@ -62,6 +62,22 @@ def test_solution_evaluation():
...
+        x[i] = m.addVar(ub=i)
+
+    for i in range(1,6):
+        m.addCons(quicksum(x[j] for j in range(20) if j%i==0) >= i)
+        m.addCons(quickprod(x[j] for j in range(20) if j%i==0) <= i**3)
+    
+    m.setObjective(quicksum(x[i] for i in range(20)))
+    m.optimize()
+    for s in m.getSols():
+        m.getSolTime(s)
```
Comment (mmghannam) [inline tests/test_solution.py:80]: There should be an assert here, to at least check that it's a positive value, and potentially check that each one has a time after the other, I'm assuming they are returned in the order they were found but I could be wrong here.

**Review by mmghannam — CHANGES_REQUESTED** [review]
Thanks a lot @Joao-Dionisio!

File: tests/test_pricer.py (hunk 1 above)
Comment (Joao-Dionisio) [reply tests/test_pricer.py:13]: I forgot to delete it, but the assertion fails, ```getDualSolVal``` returns 0, even with ```boundconstraint=True```. 

As a side not, it's annoying that sol is not capitalized in ```getDualsolLinear````

File: tests/test_reader.py (hunk 2 above)
Comment (Joao-Dionisio) [reply tests/test_reader.py]: You're right, Mo! I also noticed that there were many different spellings of sudoku in this file, I fixed those typos :)

File: tests/test_model.py [hunk 4]
```diff
@@ -170,6 +172,38 @@ def assert_conss_neq(a, b):
 
     assert_conss_neq(conss[0], conss[1])
 
+def test_addCoefLinear():
+    m = Model()
+    x = m.addVar(obj=1)
+    y = m.addVar()
```
[the remark this answers is not in the document]
Comment (Joao-Dionisio) [reply tests/test_model.py]: Done!

File: tests/test_solution.py (hunk 3 above)
Comment (Joao-Dionisio) [reply tests/test_solution.py:80]: Yeah, agreed. But unfortunately, it seems that ```getSols``` doesn't return solutions in the order they were found.

File: tests/test_solution.py [hunk 5]
```diff
@@ -79,6 +79,47 @@ def test_getSolTime():
...
+    x = m.addVar()
+    m.setObjective(x, "maximize")
+    m.setPresolve(SCIP_PARAMSETTING.OFF)
+    
+    m.optimize()
+    
+    assert m.hasPrimalRay()
+
+    m = Model()
+    x = m.addVar(lb = 0) # for readability
```
Comment (mmghannam) [inline tests/test_solution.py:93]: no need to add a comment here 😄 
```suggestion
    x = m.addVar(lb = 0)
```

File: tests/test_pricer.py (hunk 1 above)
Comment (mmghannam) [reply tests/test_pricer.py:13]:
```suggestion
```
Let's remove the line then for now. The naming comes from the corresponding SCIP C function, better to be consistent even if not the best naming.

Comment (mmghannam) [reply tests/test_pricer.py:13]: I opened a new issue #746  to track this.

**Review by mmghannam — APPROVED** [review]

**Pull request merged**